src/services/errorService.js

```diff
@@ -28,6 +31,48 @@ class ErrorService {
       Sentry.captureException(error);
     }
   }
+
+  async sanitizeError (error) {
+    // user sign rejection from metamask
+    if (error.code === 4001) {
+      return error.message;
+    }
+
+    // ignore metamask rpc header not found errors
+    if (error.code === -32000) {
+      return null;
+    }
+
+    // omg-js create UTXOs insufficient funds error
+    if (error.message && error.message.includes('Insufficient funds. Needs ')) {
+      const tokenAddress = error.message.split(' ').find(i => i.startsWith('0x'));
```

Comment:
wrap this block in a try/catch and return a generic message like "Insufficient funds to cover payments and fees" if something goes wrong here